src/s2/base/sysinfo.cc

```diff
@@ -15,15 +15,29 @@
 
 #include "s2/base/sysinfo.h"
 
-#include <sys/resource.h>
+#ifdef _MSC_VER /* if Visual C++ */
+#include <windows.h>
+#else
+#include <sys/resource.h>   // for rusage, RUSAGE_SELF
 #include <sys/time.h>
+#endif
 
 namespace base {
 
 absl::Duration CPUUsage() {
+#ifdef _MSC_VER /* if Visual C++ */
+  FILETIME creation, exit, kernel, user;
+  int res = GetProcessTimes(GetCurrentProcess(), &creation,
+                            &exit, &kernel, &user);
+  S2_CHECK_EQ(res, 0);
```

Comment:
The `CHECK` stayed in `s2testing.cc`, so just `if (res != 0) return absl::ZeroDuration();` here. (Or `if (GetProcessTimes(...) != 0) return ...;`.